pages/instructorAssessments/instructorAssessments.ejs

```diff
@@ -15,7 +16,41 @@
             trigger: 'hover',
         }).on("show.bs.popover", function() {
             $($(this).data("bs.popover").getTipElement()).css("max-width", "80%");
-        });;
+        });
+
+        var assessment_ids_needing_stats_update = <%- JSON.stringify(assessment_ids_needing_stats_update) %>;
+        async.eachLimit(assessment_ids_needing_stats_update, 3, function(assessment_id, callback) {
+          fetch('<%= urlPrefix %>/instance_admin/assessments/stats/' + assessment_id)
+            .then((response) => {
+              if (!response.ok) {
+                throw new Error('status code: ' + response.status + ', status text: ' + response.statusText);
+              }
+              return response.text();
+            }).then((text) => {
+              // parse the HTML response
+              var template = document.createElement('template');
+              template.innerHTML = text;
+              var newContent = document.importNode(template.content, true);
+
+              // update the page content
+              var row = document.getElementById('row-' + assessment_id);
+              row.querySelector('.score-stat-number').innerHTML = newContent.querySelector('.score-stat-number').innerHTML;
+              row.querySelector('.score-stat-score-hist').innerHTML = newContent.querySelector('.score-stat-score-hist').innerHTML;
+              row.querySelector('.score-stat-mean').innerHTML = newContent.querySelector('.score-stat-mean').innerHTML;
+              row.querySelector('.duration-stat-mean').innerHTML = newContent.querySelector('.duration-stat-mean').innerHTML;
+
+              // re-evaluate scripts in the row to render statistics
```

Comment:
This is kind of horrible. Rather than having scripts in each row I could have encoded the row-specific data into the HTML and then pulled the rendering code out into a top-level function. However, we don't yet have good support for client-side JS in PL (need to bring this over from PT) so it felt like it was getting out of control. This script-rehydrating code here is ugly, but at least it's fairly short and contained.

Reply:
+1 to getting better infra for client-side JS in this repo!